src/FluentMigrator/IMigrationProcessor.cs

```diff
@@ -29,8 +29,8 @@ public interface IMigrationProcessor : IQuerySchema, IDisposable
         string ConnectionString { get; }
 
         void Execute(string template, params object[] args);
-        DataSet ReadTableData(string schemaName, string tableName);
-        DataSet Read(string template, params object[] args);
+        IDataReader ReadTableData(string schemaName, string tableName);
+        IDataReader Read(string template, params object[] args);
```

Comment:
I don't know how strict we are. Do we need to rev a major version for this change?